backend/src/services/verifyToken.js

```diff
@@ -34,7 +39,30 @@ function verifyApiKey(req, res, next) {
     }
 }
 
+// verifies the google token and returns the user if valid
+function checkAuthenticated(req, res, next) {
+    let token = req.header('session-token');    // get the token from the header
+    let user = {};  // create an empty user object
+    async function verify() {
+        const ticket = await client.verifyIdToken({ // verify the token
+            idToken: token,
+            audience: process.env.CLIENT_ID,
+        });
+        const payload = ticket.getPayload();    // get the payload with the user data
+        user.name = payload.name;
+        user.email = payload.email;
+        user.picture = payload.picture;
+    }
+    verify().then(() => {
+        req.user = user; // set the user object to the request
+        next(); // call next() to continue
+    }).catch(err => {
+        res.status(401).json({ message: 'Invalid token.' })
+    });
+}
```

Comment:
this all looks good to me, but I'm having trouble getting a good response from the endpoint. I'm getting my session token by adding a console.log() that prints the access token when logging in to the mobile app (I think this is what @TienDucHo will use to call the API)
![image](https://user-images.githubusercontent.com/29143314/184995207-feadfe1a-a25b-495b-9644-752903609b66.png)


I know you use a different app to get the session token so I was wondering if it's a different token. It might just be me making the wrong request in postman though; do I need to add anything other than session-token value in the header?
![image](https://user-images.githubusercontent.com/29143314/184994895-02d14d1b-730b-479a-b902-7083ed55681f.png)